src/components/TaskOptions/TaskOptions.tsx

```diff
@@ -0,0 +1,142 @@
+import React, { useState, useContext } from 'react';
+import { useParams } from 'react-router-dom';
+import { withStyles } from '@material-ui/core/styles';
+import { UserRole, QueryNewbieArgs } from 'buddy-app-schema';
+import AuthContext, { AuthContextData } from 'contexts/AuthContext';
+import DialogContext, { DialogContextData } from 'contexts/DialogContext';
+import SnackbarContext, { SnackbarContextData } from 'contexts/SnackbarContext';
+import IconButton from '@material-ui/core/IconButton';
+import Menu from '@material-ui/core/Menu';
+import MenuItem from '@material-ui/core/MenuItem';
+import MoreVertIcon from '@material-ui/icons/MoreVert';
+import Fade from '@material-ui/core/Fade';
+import ListItemIcon from '@material-ui/core/ListItemIcon';
+import ListItemText from '@material-ui/core/ListItemText';
+import EditIcon from '@material-ui/icons/Edit';
+import FileCopyIcon from '@material-ui/icons/FileCopy';
+import DeleteIcon from '@material-ui/icons/Delete';
+import DICTIONARY from './dictionary';
+import { TaskOptionsProps } from './types';
+
+const StyledMenu = withStyles(theme => ({
+  paper: {
+    backgroundColor: theme.palette.background.default,
+    boxShadow: theme.shadows[2],
+  },
+}))(Menu);
+
+const StyledListItemIcon = withStyles(theme => ({
+  root: {
+    minWidth: 'auto',
+    paddingRight: theme.spacing(1.5),
+    [theme.breakpoints.down('xs')]: {
+      display: 'none',
+    },
+  },
+}))(ListItemIcon);
+
+const TaskOptions: React.FC<TaskOptionsProps> = ({
+  id: taskId,
+  taskOptionHandlers = {},
+}) => {
+  const { deleteTask } = taskOptionHandlers;
+  const {
+    data: { role },
+  } = useContext<AuthContextData>(AuthContext);
+  const { showDialog, hideDialog } = useContext<DialogContextData>(DialogContext);
+  const { showSnackbar } = useContext<SnackbarContextData>(SnackbarContext);
+  const { newbieId } = useParams<QueryNewbieArgs>();
+  const [anchorEl, setAnchorEl] = useState<null | HTMLElement>(null);
+  const isOpened = Boolean(anchorEl);
+
+  const handleClick = (e: React.MouseEvent<HTMLButtonElement>) =>
+    setAnchorEl(e.currentTarget);
+
+  const handleClose = () => setAnchorEl(null);
+
+  const handleDelete = () =>
+    showDialog(
+      DICTIONARY.DELETE_DIALOG.MESSAGE,
+      DICTIONARY.DELETE_DIALOG.TITLE,
+      async () => {
+        handleClose();
+        hideDialog();
+
+        try {
+          deleteTask && (await deleteTask({ variables: { taskId, newbieId } }));
+          showSnackbar(DICTIONARY.DELETE_SNACKBAR.SUCCESS);
+        } catch (err) {
+          showSnackbar(DICTIONARY.DELETE_SNACKBAR.ERROR);
+        }
+      }
+    );
+
+  const options = [
```

Comment:
Maybe it's a good idea to define types for the option object.

Reply:
Fixed here [#6f88e07](https://github.com/BuildIt-Poland/buddy-app/pull/53/commits/6f88e078eb6ab5479d1ffc702b6e5ceea736b954#diff-0fd0bc4acdcd9be9127919bea5cd012c)